### PR TITLE
chore: add STEP_WITH_OPTIMIZER setting for lr scheduler

### PR DESCRIPTION
**Pull request by stoksc**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: harness/determined/pytorch/_lr_scheduler.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ class StepMode(enum.Enum):
         STEP_EVERY_EPOCH = 1
         STEP_EVERY_BATCH = 2
         MANUAL_STEP = 3
+        STEP_EVERY_OPTIMIZER_STEP = 4
```
Comment (rb-determined-ai) [inline harness/determined/pytorch/_lr_scheduler.py:31]: Looks like you need to add this to the docstring

**Review by rb-determined-ai — COMMENTED** [review]
I think this is a significant enough improvement we should be recommending this mode over `STEP_EVERY_BATCH` in most cases.  Can you add some docs somewhere?

File: harness/determined/pytorch/_lr_scheduler.py [hunk 2]
```diff
@@ -49,7 +51,15 @@ def __init__(
                 2. ``STEP_EVERY_BATCH``: Determined will call scheduler.step() after every
                    ``frequency`` training batch(es). No arguments will be passed to step().
 
-                3. ``MANUAL_STEP``: Determined will not call scheduler.step() at all.
+                3. ``STEP_EVERY_OPTIMIZER_STEP``: Determined will call scheduler.step() in sync
```
Comment (rb-determined-ai) [inline harness/determined/pytorch/_lr_scheduler.py:54]: Can we alter the docs for ``STEP_EVERY_BATCH`` to suggest that ``STEP_EVERY_OPTIMIZER_STEP`` should be preferred?

Comment (stoksc) [reply harness/determined/pytorch/_lr_scheduler.py:54]: sure, makes sense.

**Review by rb-determined-ai — APPROVED** [review]
